Libplanet/Crypto/BlsPrivateKey.cs

```diff
@@ -0,0 +1,280 @@
+        /// above all cases are not true, then use <see cref="BlsSignature.MultiVerify"/> without a
+        /// signature aggregation.</description></item>
+        /// </list>
+        /// </summary>
+        /// <param name="message">A message <see cref="byte"/>s to sign.</param>
+        /// <returns>A signature that proves the authenticity of the <paramref name="message"/>.
+        /// It can be verified using <see cref="Libplanet.Crypto.BlsPublicKey.Verify"/>,
+        /// <see cref="BlsSignature.AggregateVerify"/>,
+        /// <see cref="BlsSignature.FastAggregateVerify"/>, or
+        /// <see cref="BlsSignature.MultiVerify"/> method with requiring use-case.
+        /// </returns>
+        /// <seealso cref="Libplanet.Crypto.BlsPublicKey.Verify"/>
+        /// <seealso cref="BlsSignature.AggregateVerify"/>
+        /// <seealso cref="BlsSignature.FastAggregateVerify"/>
+        /// <seealso cref="BlsSignature.MultiVerify"/>
+        public byte[] Sign(byte[] message)
+        {
+            HashDigest<SHA256> hashed = HashDigest<SHA256>.DeriveFrom(message);
+            return CryptoConfig.ConsensusCryptoBackend.Sign(hashed, this);
+        }
+
+        /// <inheritdoc cref="Sign(byte[])"/>
+        public byte[] Sign(ImmutableArray<byte> message) => Sign(message.ToArray());
+
+        /// <summary>
+        /// Encodes the private key into a corresponding mutable <see cref="byte"/> array
+        /// representation.
+        /// </summary>
+        /// <returns>An encoded <see cref="byte"/> array representation.  It guarantees that
+        /// returned arrays are never reused, and mutating on them does not affect
+        /// <see cref="BlsPrivateKey"/> instance's internal states.</returns>
+        /// <remarks>
+        /// An encoded <see cref="byte"/> array representation can be recovered to a <see
+        /// cref="BlsPrivateKey"/> instance again using
+        /// <see cref="BlsPrivateKey(IReadOnlyList{byte})"/> constructor.
+        /// <para>As like <see cref="BlsPrivateKey"/> instances, it also must be kept secret.
+        /// In practice, this must not be sent over the network, and be securely stored in the file
+        /// system.
+        /// </para>
+        /// <para>To get an immutable array instead of mutable one, use <see cref="ByteArray"/>
+        /// property.</para>
+        /// </remarks>
+        /// <seealso cref="ByteArray"/>
+        /// <seealso cref="BlsPrivateKey(IReadOnlyList{byte})"/>
+        [Pure]
+        public byte[] ToByteArray() => _privateKey.ToArray();
+
+        private static byte[] GenerateBytesFromHexString(string hex)
+        {
+            byte[] bytes = ByteUtil.ParseHex(hex);
+            if (bytes.Length != KeyByteSize)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(hex),
+                    $"Expected {KeyByteSize * 2} hexadecimal digits."
+                );
+            }
+
+            return bytes;
```

Comment:
```suggestion
            return bytes.Length == KeyByteSize
                ? bytes
                : throw new ArgumentOutOfRangeException(
                    nameof(hex),
                    $"Expected {KeyByteSize * 2} hexadecimal digits.");
```

Reply:
Could this change be postponable? I'd like to keep the code style with `PrivateKey`.